chat/src/test/java/greencity/service/impl/ChatRoomServiceImplTest.java

```diff
@@ -0,0 +1,126 @@
+            .email(email)
+            .profilePicture(null)
+            .userStatus(UserStatus.ACTIVATED)
+            .build();
+        expectedSet.add(expectedParticipant);
+        expected = ChatRoom.builder()
+            .id(1L)
+            .name("test")
+            .messages(new LinkedList<>())
+            .type(ChatType.PRIVATE)
+            .participants(new HashSet<>())
+            .build();
+        expectedList.add(expected);
+        expectedToReturn = ChatRoom.builder()
+            .name("chatName")
+            .participants(expectedSet)
+            .type(ChatType.PRIVATE)
+            .build();
+        expectedDto = ChatRoomDto.builder()
+            .id(1L)
+            .name("test")
+            .messages(new LinkedList<>())
+            .chatType(ChatType.PRIVATE)
+            .participants(new HashSet<>())
+            .build();
+    }
+
+    @Test
+    void findAllByParticipantName() {
+        when(participantService.findByEmail(email)).thenReturn(expectedParticipant);
+        when(chatRoomRepo.findAllByParticipant(expectedParticipant)).thenReturn(expectedList);
+        when(modelMapper.map(expectedList, new TypeToken<List<ChatRoomDto>>() {
+        }.getType())).thenReturn(expectedListDto);
+        List<ChatRoomDto> actual = chatRoomService.findAllByParticipantName(email);
+        assertEquals(expectedListDto, actual);
+    }
+
+    @Test
+    void findAllRoomsByParticipantsAndStatus() {
+        when(chatRoomRepo.findByParticipantsAndStatus(expectedSet, expectedSet.size(), ChatType.GROUP))
+            .thenReturn(expectedList);
+        when(modelMapper.map(expectedList, new TypeToken<List<ChatRoomDto>>() {
+        }.getType())).thenReturn(expectedListDto);
+        List<ChatRoomDto> actual = chatRoomService.findAllRoomsByParticipantsAndStatus(expectedSet, ChatType.GROUP);
+        assertEquals(expectedListDto, actual);
+    }
+
+    @Test
+    void findChatRoomById() {
+        when(chatRoomRepo.findById(1L)).thenReturn(Optional.of(expected)).thenThrow(ChatRoomNotFoundException.class);
+        when(modelMapper.map(expected, ChatRoomDto.class)).thenReturn(expectedDto);
+        ChatRoomDto actual = chatRoomService.findChatRoomById(1L);
+        assertEquals(expectedDto, actual);
+    }
+
+    @Test
+    void findPrivateByParticipants() {
+        when(participantService.findByEmail(email)).thenReturn(expectedParticipant);
+        when(participantService.findById(1L)).thenReturn(expectedParticipant);
+        when(chatRoomRepo.findByParticipantsAndStatus(expectedSet, expectedSet.size(), ChatType.PRIVATE))
+            .thenReturn(expectedListEmpty);
+        when(chatRoomRepo.save(expectedToReturn)).thenReturn(expectedToReturn);
+        when(modelMapper.map(expectedToReturn, ChatRoomDto.class)).thenReturn(expectedDto);
+        ChatRoomDto actualExpectedListEmpty = chatRoomService.findPrivateByParticipants(1L, email);
+        assertEquals(expectedDto, actualExpectedListEmpty);
```

Comment:
Below is the same condition. Is it needed here?

Reply:
Yes, one condition - with an empty list, and the other - with
filled list.